File: src/main/java/rars/venus/settings/editor/EditorSettingsPanel.java
package rars.venus.settings.editor;

import org.jetbrains.annotations.NotNull;

import javax.swing.*;
import javax.swing.plaf.basic.BasicSplitPaneDivider;
import javax.swing.plaf.basic.BasicSplitPaneUI;
import java.awt.*;

public final class EditorSettingsPanel extends JSplitPane {
    public EditorSettingsPanel(final @NotNull TreePanel left, final @NotNull JComponent right) {
        super(JSplitPane.HORIZONTAL_SPLIT, left, right);
        setDividerLocation(0.5);
        setMaximumSize(new Dimension(Integer.MAX_VALUE, Integer.MAX_VALUE));
        setDividerSize(2);
        setUI(new PlainSplitPaneDividerUI());
        setBorder(BorderFactory.createEmptyBorder(10, 10, 10, 10));
    }

    /**
     * Custom divider UI that removes the dots from the divider.
     */
    private static class PlainSplitPaneDividerUI extends BasicSplitPaneUI {
        @Override
        public BasicSplitPaneDivider createDefaultDivider() {
            return new BasicSplitPaneDivider(this) {
                @Override
                public void paint(Graphics g) {
                    // Remove painting of any dots by leaving this empty
                    g.setColor(splitPane.getBackground());
                    g.fillRect(0, 0, getWidth(), getHeight());
                }
            };
        }
    }
}

File: src/main/java/rars/venus/settings/editor/EditorSettingsPanel.kt
package rars.venus.settings.editor

import rars.venus.settings.editor.views.PanelWithTextAreaView
import rars.venus.util.BorderLayout
import java.awt.BorderLayout
import java.awt.Dimension
import java.awt.Graphics
import javax.swing.BorderFactory
import javax.swing.JPanel
import javax.swing.JSplitPane
import javax.swing.plaf.basic.BasicSplitPaneDivider
import javax.swing.plaf.basic.BasicSplitPaneUI

class EditorSettingsPanel(
    treePanel: TreePanel,
    val panelWithTextAreaView: PanelWithTextAreaView
) : JPanel() {
    val bottomRowComponent = BottomRowComponent().apply {
        border = BorderFactory.createEmptyBorder(10, 0, 0, 0)
    }

    init {
        border = BorderFactory.createEmptyBorder(10, 10, 10, 10)
        BorderLayout {
            this[BorderLayout.CENTER] = buildMainPart(treePanel, panelWithTextAreaView)
            this[BorderLayout.SOUTH] = bottomRowComponent
        }
    }
}

private fun buildMainPart(
    treePanel: TreePanel,
    panelWithTextAreaView: PanelWithTextAreaView
): JSplitPane = JSplitPane(JSplitPane.HORIZONTAL_SPLIT, treePanel, panelWithTextAreaView).apply {
    setDividerLocation(0.5)
    maximumSize = Dimension(Int.Companion.MAX_VALUE, Int.Companion.MAX_VALUE)
    dividerSize = 2
    setUI(PlainSplitPaneDividerUI)
}

/**
 * Custom divider UI that removes the dots from the divider.
 */
object PlainSplitPaneDividerUI : BasicSplitPaneUI() {
    override fun createDefaultDivider(): BasicSplitPaneDivider = PlainSplitPaneDivider(this)
}

private class PlainSplitPaneDivider(ui: BasicSplitPaneUI) : BasicSplitPaneDivider(ui) {
    override fun paint(g: Graphics) {
        // Remove painting of any dots by leaving this empty
        g.color = splitPane.background
        g.fillRect(0, 0, width, height)
    }
}